api/src/main/java/jakarta/servlet/SessionCookieConfig.java

```diff
@@ -231,4 +233,45 @@
      * @see jakarta.servlet.http.Cookie#getMaxAge
      */
     public int getMaxAge();
+
+    /**
+     * Sets the value for the given session cookie attribute. When a value is set via this method, the value returned by the
+     * attribute specific getter (if any) must be consistent with the value set via this method.
+     *
+     * @param name Name of attribute to set
+     * @param value Value of attribute
+     *
+     * @throws IllegalStateException if the associated ServletContext has already been initialised
+     *
+     * @throws IllegalArgumentException If the attribute name is null or contains any characters not permitted for use in
+     * Cookie names.
+     *
+     * @throws NumberFormatException If the attribute is known to be numerical but the provided value cannot be parsed to a
+     * number.
+     *
+     * @since Servlet 6.0
+     */
+    public void setAttribute(String name, String value);
+
+    /**
+     * Obtain the value for a given session cookie attribute. Values returned from this method must be consistent with the
+     * values set and returned by the attribute specific getters and setters in this class.
+     *
+     * @param name Name of attribute to return
+     *
+     * @return Value of specified attribute
+     *
+     * @since Servlet 6.0
+     */
+    public String getAttribute(String name);
+
+    /**
+     * Obtain the Map (keys are case insensitive) of all attributes and values, including those set via the attribute
```

Comment:
If the keys are case-insensitive (must they be?) then that needs to be mentioned in the javadoc of both `setAttribute` and `getAttribute`.

Rather than case-insensitive, could we just convert all the keys to lowercase?

Reply:
Done.